### PR TITLE
deposit-form: validate saved draft on form load

### PR DESCRIPTION
**Pull request by audrium**
closes https://github.com/inveniosoftware/invenio-app-rdm/issues/737

Validates saved draft when the form initially loads to show any errors and allow publishing on page reload or edit. Result:
pr4

### CODE REVIEW

**Review by fenekku — APPROVED** [review]
Mostly renaming and it's good to go. 

Maybe we talked about this (I hope!), but the other avenue which might be more involved would be to have the backend do a preliminary save when a draft is edited and provide the draft with errors to frontend. The advantage being a reduction in API calls. The FormFeedback would have to be modified to only show errors in this case (and no saved successfully). Might be something for the future...

File: src/lib/DepositBootstrap.js [hunk 1]
```diff
@@ -32,6 +41,7 @@ class DepositBootstrapComponent extends Component {
         formik={{
           enableReinitialize: true,
           initialValues: this.props.record,
+          innerRef: node => (this.formik = node)
```
[the remark this answers is not in the document]
Comment (audrium) [reply src/lib/DepositBootstrap.js:44]: Actually no, this is used [here](https://github.com/inveniosoftware/react-invenio-deposit/blob/4e8acd3f0c33d73ba744a157838eeddc1d422ccd/src/lib/DepositBootstrap.js#L21) to pass `formik` object to `submitForm` function and eventually to `save` action

File: src/lib/state/types/index.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ export const FORM_ACTION_EVENT_EMITTED = 'FORM_ACTION_EVENT_EMITTED';
 export const FORM_ACTION_FAILED = 'FORM_ACTION_FAILED';
 export const FORM_INITIALIZED = "FORM_INITIALIZED";
 export const FORM_SAVING = "FORM_SAVING";
+export const FORM_VALIDATING = "FORM_VALIDATING";
+export const FORM_VALIDE_SUCCEEDED = "FORM_VALIDE_SUCCEEDED";
```
[the remark this answers is not in the document]
Comment (audrium) [reply src/lib/state/types/index.js]: thanks, updated the code with your solution

**Comment by audrium** [comment]
> Mostly renaming and it's good to go.
> 
> Maybe we talked about this (I hope!), but the other avenue which might be more involved would be to have the backend do a preliminary save when a draft is edited and provide the draft with errors to frontend. The advantage being a reduction in API calls. The FormFeedback would have to be modified to only show errors in this case (and no saved successfully). Might be something for the future...

Yes, we indeed talked about it! And decided to take this approach for now and leave the backend implementation for the future as you mentioned :)

**Pull request closed without merging**